main/src/ca/uwaterloo/flix/language/CompilationError.scala

```diff
@@ -30,6 +31,11 @@ trait CompilationError {
     */
   def kind: String
 
+  /**
+    * Returns the severity of the error.
+    */
+  def severity: Severity = Severity.Error
```

Comment:
Generally I am not a fan of overriding fields in super classes. Arguably  violating the LSP here.

Reply:
Agreed. The alternative is to leave it abstract and implement it in every single error.

That is probably better and could be done in the future (or now)

Reply:
Either that or split into two abstract classes or subtraits, one for errors and one for hints.